From 8aa0d83749609ed571b984d2dc64c7f757950c7c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 1 Dec 2021 12:10:11 -0800
Subject: [PATCH v2] backtrace: Include cxxabi.h without HAVE_DW

Since its used in code without HAVE_DW, it fails to compile on such
systems e.g. linux/musl

Fixes
src/libcamera/base/backtrace.cpp:235:16: error: use of undeclared identifier 'abi'
                char *name = abi::__cxa_demangle(symbol, nullptr, nullptr, nullptr);
                             ^
1 error generated.

Upstream-Status: Submitted [https://lists.libcamera.org/pipermail/libcamera-devel/2021-December/027989.html]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/libcamera/base/backtrace.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libcamera/base/backtrace.cpp b/src/libcamera/base/backtrace.cpp
index d93e5518..42c60f7c 100644
--- a/src/libcamera/base/backtrace.cpp
+++ b/src/libcamera/base/backtrace.cpp
@@ -13,11 +13,12 @@
 #endif
 
 #ifdef HAVE_DW
-#include <cxxabi.h>
 #include <elfutils/libdwfl.h>
 #include <unistd.h>
 #endif
 
+#include <cxxabi.h>
+
 #if HAVE_UNWIND
 /*
  * Disable support for remote unwinding to enable a more optimized
-- 
2.34.1

